compose/config/config.py

```diff
@@ -430,6 +431,12 @@ def resolve_environment(service_dict):
     return dict(resolve_env_var(k, v) for k, v in six.iteritems(env))
 
 
+def resolve_build_args(build):
+    args = {}
+    args.update(parse_build_arguments(build.get('args')))
```

Comment:
I think since `parse_build_arguments()` always returns a new dict, these two lines could be:

```python
args = parse_build_arguments(build.get('args'))
```